client-mr/src/main/java/org/apache/hadoop/mapreduce/v2/app/RssMRAppMaster.java

```diff
@@ -221,13 +189,64 @@ public Thread newThread(Runnable r) {
         conf.setInt(MRJobConfig.REDUCE_MAX_ATTEMPTS, originalAttempts + inc);
       }
 
-      LOG.info("Start to register shuffle");
-      long start = System.currentTimeMillis();
-      serverToPartitionRanges.entrySet().forEach(entry -> {
-        client.registerShuffle(
-            entry.getKey(), appId, 0, entry.getValue(), remoteStorage);
+      // retryInterval must bigger than `rss.server.heartbeat.timeout`, or maybe it will return the same result
```

Comment:
Could the `retryInterval` be configuration option? And check this configuration option like
```
if (retryInterval < heartbeatTimeout) {
   LOG.warn("xxx");
}
```